bin/stacks

```diff
@@ -0,0 +1,333 @@
+
+
+@throttling_retry
+def get_vpc_id(conn, name):
+    '''Returns the first VPC ID given its name and region'''
+    vpcs = conn.get_all_vpcs(filters={'tag:Name': name})
+    conn.close()
+    if len(vpcs) == 1:
+        return vpcs[0].id
+    else:
+        raise RuntimeError('{} VPC not found'.format(name))
+
+
+@throttling_retry
+def get_stack_output(conn, name, key):
+    '''Returns stack output key value'''
+    result = conn.describe_stacks(name)
+    if len(result) != 1:
+        raise RuntimeError('{} stack not found'.format(name))
+    outputs = [s.outputs for s in result][0]
+    for output in outputs:
+        if output.key == key:
+            return output.value
+    raise RuntimeError('{} output not found'.format(key))
+
+
+@throttling_retry
+def get_stack_tag(conn, name, tag):
+    '''Returns stack tag'''
+    result = conn.describe_stacks(name)
+    if len(result) != 1:
+        raise RuntimeError('{} stack not found'.format(name))
+    tags = [s.tags for s in result][0]
+    return tags.get(tag, '')
+
+
+def gen_template(template, config, pretty=False):
+    '''Return generated CloudFormation template string'''
+    tpl = jinja2.Template(template.read())
+    out = tpl.render(config)
+    yaml_out = yaml.load(out)
+    indent = 2 if pretty else None
+    return json.dumps(yaml_out, indent=indent)
+
+
+def list_stacks(conn, name_filter='*', verbose=False):
+    '''Lists active stacks'''
+    stacks_filters = [
+        'CREATE_COMPLETE',
+        'CREATE_IN_PROGRESS',
+        'CREATE_FAILED',
+        'DELETE_IN_PROGRESS',
+        'DELETE_FAILED',
+        'ROLLBACK_COMPLETE',
+        'ROLLBACK_FAILED',
+        'ROLLBACK_IN_PROGRESS',
+        'UPDATE_COMPLETE',
+        'UPDATE_IN_PROGRESS',
+        'UPDATE_ROLLBACK_COMPLETE',
+        'UPDATE_ROLLBACK_FAILED',
+        'UPDATE_ROLLBACK_IN_PROGRESS',
+    ]
+
+    s = conn.list_stacks(stacks_filters)
+
+    stacks = []
+    for n in s:
+        if name_filter and _match_stack_name(n.stack_name, name_filter):
+            columns = [n.stack_name, n.stack_status]
+            if verbose:
+                env = get_stack_tag(conn, n.stack_name, 'Env')
+                columns.append(env)
+                columns.append(n.template_description)
+            stacks.append(columns)
+            columns = []
+
+    if len(stacks) >= 1:
+        return tabulate(stacks, tablefmt='plain')
+    else:
+        return 'No stacks found'
+
+
+def create_stack(conn, name, template, config, update=False, dry=False):
+    '''Creates or updates CloudFormation stack from a jinja2 template'''
+    tags = {
+        'Env': config['env'],
+    }
+
+    tpl = gen_template(template, config, dry)
+    if dry:
+        print(tpl)
+        print("Template size: " + str(len(tpl)), file=sys.stderr)
+        return True
+
+    bucket_name = '{}-stacks-{}'.format(config['env'], config['region'])
+    b = config['s3_conn'].get_bucket(bucket_name)
```

Comment:
Sure. Will refactor that bit. I change a little bit, so that different regions are supported now.